backend/compact-connect/lambdas/provider-data-v1/tests/unit/test_sqs_handler.py

```diff
@@ -0,0 +1,83 @@
+import json
+from unittest.mock import Mock
+from uuid import uuid4
+
+from tests import TstLambdas
+
+
+class TestSQSHandler(TstLambdas):
+    def test_happy_path(self):
+        from handlers.utils import sqs_handler
+
+        @sqs_handler
+        def message_handler(message: dict):  # pylint: disable=unused-argument
+            return
+
+        event = {
+            'Records': [
+                {
+                    'messageId': str(uuid4()),
+                    'body': json.dumps({'foo': 'bar'})
+                }
+            ]
+        }
+
+        resp = message_handler(event, self.mock_context)  # pylint: disable=too-many-function-args
+
+        self.assertEqual(
+            {'batchItemFailures': []},
+            resp
+        )
+
+    def test_partial_failure(self):
+        from handlers.utils import sqs_handler
+
+        mock_partial_failures = Mock(
+            # Responses when called - three successes, two failures
+            side_effect=[
+                None,
+                RuntimeError('Oh no!'),
+                None,
+                None,
+                RuntimeError('Not again!')
+            ]
+        )
+
+        @sqs_handler
+        def message_handler(message: dict):  # pylint: disable=unused-argument
+            return mock_partial_failures()
```

Comment:
We're taking advantage of the magic of `Mock` objects here. When we define `side_effect` as an Iterable, the `Mock` object will conveniently iterate through the objects in the Iterable, returning each one in order. In this case, we're getting five different responses across five different calls. Leveraging this feature saves us the effort of writing some sort of generator/coroutine that accomplishes the same thing.

Reply:
Ah, makes sense. I read this incorrectly as the mock side effect returning the list once, instead of returning those 5 items over the course of 5 function calls.